apps/intellij/src/main/kotlin/dev/nx/console/toolWindow/NxProjectsTreeStructure.kt

```diff
@@ -0,0 +1,327 @@
+        }
+
+        class Tasks(private val nxWorkspace: NxWorkspace, parent: SimpleNode) :
+            NxSimpleNode(parent) {
+            init {
+                icon = AllIcons.Nodes.ConfigFolder
+            }
+
+            override fun buildChildren(): Array<SimpleNode> =
+                nxWorkspace.workspace.projects.values
+                    .flatMap { p -> p.targets.keys.map { it to p.name } }
+                    .groupBy { it.first }
+                    .map { Task(nxWorkspace, it.key, this) }
+                    .toTypedArray()
+
+            override fun getName(): String = "Tasks"
+        }
+
+        class Task(private val nxWorkspace: NxWorkspace, val taskName: String, parent: SimpleNode) :
+            NxSimpleNode(parent) {
+            init {
+                icon = AllIcons.Nodes.ConfigFolder
+            }
+
+            override fun buildChildren(): Array<SimpleNode> =
+                nxWorkspace.workspace.projects
+                    .filter { it.value.targets.contains(taskName) }
+                    .map {
+                        NxTarget(
+                            name = it.key,
+                            nxProject = it.key,
+                            nxTarget = taskName,
+                            parent = this
+                        )
+                    }
+                    .toTypedArray()
+
+            override fun getName(): String = taskName
+        }
+
+        class NxTarget(
+            private val name: String,
+            val nxProject: String,
+            val nxTarget: String,
+            parent: SimpleNode
+        ) : NxSimpleNode(parent) {
+            init {
+                icon = AllIcons.General.Gear
+            }
+
+            override fun buildChildren(): Array<SimpleNode> = emptyArray()
+
+            override fun getName(): String = name
+        }
+
+        class Project(val nxProject: NxProject, parent: SimpleNode) : NxSimpleNode(parent) {
+
+            init {
+                icon =
+                    if (nxProject.projectType == "application") AllIcons.Nodes.Module
+                    else AllIcons.Nodes.PpLib
+            }
+
+            override fun buildChildren(): Array<SimpleNode> =
+                nxProject.targets.keys
+                    .map {
+                        NxTarget(
+                            name = it,
+                            nxProject = nxProject.name,
+                            nxTarget = it,
+                            parent = this
+                        )
+                    }
+                    .toTypedArray()
+
+            override fun getName(): String = nxProject.name
+        }
+    }
+
+    private inner class RunAction : ExecuteAction(DefaultRunExecutor.getRunExecutorInstance()) {
+        init {
+            shortcutSet = CommonShortcuts.ENTER
+        }
+    }
+
+    private inner class EditRunSettingsAction : AnAction(AllIcons.Actions.EditSource), DumbAware {
```

Comment:
What do you think about splitting some of these classes out into separate files? I think having a 300 line file is not very readable and it would be easier to explore if for example the action code was separate.

Reply:
They are inner classes they access to fields of the outer class and they are not intended to be used outside...
I saw this pattern while navigating on intellij code base

Reply:
They are inner classes, they access outer class fields like the tree to reload it for example...

Reply:
yeah that makes sense... let's leave it like this for now